src/s3path.jl

```diff
@@ -193,6 +193,63 @@ function Base.isdir(fp::S3Path)
     return iterate(objects) !== nothing
 end
 
+function FilePathsBase.walkpath(fp::S3Path; kwargs...)
+    # Select objects with that prefix
+    objects = s3_list_objects(fp.config, fp.bucket, fp.key; delimiter="")
+
+    # Construct a new Channel using a recursive internal `_walkpath!` function
+    return Channel(ctype=typeof(fp)) do chnl
+        _walkpath!(fp, fp, Iterators.Stateful(objects), chnl; kwargs...)
+    end
+end
+
+function _walkpath!(root::S3Path, prefix::S3Path, objects, chnl; topdown=true, onerror=throw, kwargs...)
```

Comment:
Gripe for FilePathsBase. It would be good to have documentation on `walkpath` for its keyword arguments

Reply:
Yeah, it's a carry over from `walkdir`. I'm not entirely sure how to test those without relying on mocking though.